Bug: Typescript issue

### Bug description

Following installation of the node_module (8.5.0) in a remix application and running `tsc --noEmit` I am getting a TS error from `node_modules/react-day-picker/dist/index.d.t`.

The problem is with `"onResize" | "onResizeCapture"` both of which are not assignable to `ButtonProps`

```
node_modules/react-day-picker/dist/index.d.ts:902:75 - error TS2344: Type '"max" | "required" | "type" | "data" | "default" | "high" | "low" | "key" | "id" | "media" | "height" | "width" | "start" | "open" | "name" | "color" | "content" | "translate" | "value" | ... 341 more ... | "onResizeCapture"' does not satisfy the constraint 'keyof ButtonProps'.
  Type '"onResizeCapture"' is not assignable to type 'keyof ButtonProps'. Did you mean '"onResetCapture"'?
```

<img width="561" alt="Screenshot 2023-01-27 at 10 32 07" src="https://user-images.githubusercontent.com/3248857/215065396-29bda3b1-868a-4544-98bc-212814f83170.png">

<img width="1402" alt="Screenshot 2023-01-27 at 10 36 47" src="https://user-images.githubusercontent.com/3248857/215066415-636f4974-6a47-4b55-a573-26039466c1fc.png">

--- a/packages/react-day-picker/src/components/Button/Button.tsx
+++ b/packages/react-day-picker/src/components/Button/Button.tsx
@@ -3,11 +3,9 @@ import React, { forwardRef } from 'react';
 import { useDayPicker } from 'contexts/DayPicker';
 
 /** The props for the {@link Button} component. */
-export type ButtonProps = React.HTMLProps<HTMLButtonElement>;
+export type ButtonProps = JSX.IntrinsicElements['button'];
 
-/**
- * Render a button HTML element applying the reset class name.
- */
+/** Render a button HTML element applying the reset class name. */
 export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   (props, ref) => {
     const { classNames, styles } = useDayPicker();

